libpod/image/image.go

```diff
@@ -131,41 +142,67 @@ func (ir *Runtime) NewFromLocal(name string) (*Image, error) {
 	return &image, nil
 }
 
+// DefaultNewImageConfig returns a default new image config for runtime.New()
+func (ir *Runtime) DefaultNewImageConfig(name string) *NewImageConfig {
+	imgCtx := new(NewImageConfig)
+	imgCtx.Name = name
+	imgCtx.SignaturePolicyPath = ir.SignaturePolicyPath
+	imgCtx.Authfile = ""
+	imgCtx.Writer = nil
+	imgCtx.DockerOptions = nil
+	imgCtx.SigningOptions = SigningOptions{}
+	imgCtx.Label = nil
+	return imgCtx
+}
+
 // New creates a new image object where the image could be local
 // or remote
-func (ir *Runtime) New(ctx context.Context, name, signaturePolicyPath, authfile string, writer io.Writer, dockeroptions *DockerRegistryOptions, signingoptions SigningOptions, forcePull bool, label *string) (*Image, error) {
+func (ir *Runtime) New(ctx context.Context, config *NewImageConfig, forcePull, update bool) (*Image, error) {
 	span, _ := opentracing.StartSpanFromContext(ctx, "newImage")
 	span.SetTag("type", "runtime")
 	defer span.Finish()
 
 	// We don't know if the image is local or not ... check local first
 	newImage := Image{
-		InputName:    name,
+		InputName:    config.Name,
 		Local:        false,
 		imageruntime: ir,
 	}
+
+	var imageName []string
+	var err error
+
 	if !forcePull {
 		localImage, err := newImage.getLocalImage()
+		if update {
+			if err != nil {
+				return nil, errors.Wrapf(err, "Unable to update an image that isn't in local storage")
```

Comment:
Does this make sense? I would think that update would pull the new image in this case without error